### PR TITLE
worked on customers CRUD, updated docs

### PR DESCRIPTION
**Pull request by Psami-wondah**

### CODE REVIEW

File: bigfastapi/customer.py [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+from fastapi import status
+
+app = APIRouter(tags=["Customers 💁"],)
+
+
+
+
+@app.post("/customers", response_model=customer_schemas.CustomerCreateResponse)
+def create_customer(customer: customer_schemas.CustomerValidator, db: orm.Session = fastapi.Depends(get_db)):
+    organization = db.query(organisation_models.Organization).filter(organisation_models.Organization.name == customer.organization_name).first()
```
Comment (markessien) [inline bigfastapi/customer.py:22]: You cannot use organisation name. ID, obviously

File: bigfastapi/customer.py [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+        db.add(customer_instance)
+        db.commit()
+        db.refresh(customer_instance)
+        return {"message": "Customer created succesfully", "customer": customer_schemas.CustomerInDB.from_orm(customer_instance)}
+    else:
+        return {"message": "Organization Doesn't Exist", "customer": {}}
+
+@app.get('/customers', response_model=List[customer_schemas.CustomerInDB])
+def get_customers(db: orm.Session = fastapi.Depends(get_db)):
+    customers = db.query(customer_models.Customer).all()
```
Comment (markessien) [inline bigfastapi/customer.py:53]: What if there are 10 million customers? You return them all?

**Review by markessien — CHANGES_REQUESTED** [review]

File: bigfastapi/schemas/customer_schemas.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+class CustomerValidator(Customer):
+    organization_name: str
+
+class CustomerInDB(Customer):
+    organization: str
+    customer_id: str
+    date_created: datetime
+    last_updated: datetime
+
+class CustomerUpdate(BaseModel):
```
[the remark this answers is not in the document]
Comment (Psami-wondah) [reply bigfastapi/schemas/customer_schemas.py:32]: It is different from the customer. It has optional fields which are None by default. It is for the update endpoint only

File: bigfastapi/customer.py (hunk 1 above)
Comment (Psami-wondah) [reply bigfastapi/customer.py:22]: fixed

File: bigfastapi/customer.py [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+
+        )
+        db.add(customer_instance)
+        db.commit()
+        db.refresh(customer_instance)
+        return {"message": "Customer created succesfully", "customer": customer_schemas.CustomerInDB.from_orm(customer_instance)}
+    else:
+        return {"message": "Organization Doesn't Exist", "customer": {}}
+
+@app.get('/customers', response_model=List[customer_schemas.CustomerInDB])
```
[the remark this answers is not in the document]
Comment (Psami-wondah) [reply bigfastapi/customer.py:51]: added auth

File: bigfastapi/customer.py (hunk 2 above)
Comment (Psami-wondah) [reply bigfastapi/customer.py:53]: Added pagination

File: bigfastapi/customer.py [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+def get_customers(db: orm.Session = fastapi.Depends(get_db)):
+    customers = db.query(customer_models.Customer).all()
+    return list(map(customer_schemas.CustomerInDB.from_orm, customers))
+
+@app.get('/customers/organization/{organization_name}', response_model=List[customer_schemas.CustomerInDB])
+def get_customers_by_organization(organization_name: str, db: orm.Session = fastapi.Depends(get_db)):
+    organization = db.query(organisation_models.Organization).filter(organisation_models.Organization.name == organization_name).first()
```
[the remark this answers is not in the document]
Comment (Psami-wondah) [reply bigfastapi/customer.py:58]: I don't understand

File: bigfastapi/customer.py [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+        return {"message": "Customer created succesfully", "customer": customer_schemas.CustomerInDB.from_orm(customer_instance)}
+    else:
+        return {"message": "Organization Doesn't Exist", "customer": {}}
+
+@app.get('/customers', response_model=List[customer_schemas.CustomerInDB])
+def get_customers(db: orm.Session = fastapi.Depends(get_db)):
+    customers = db.query(customer_models.Customer).all()
+    return list(map(customer_schemas.CustomerInDB.from_orm, customers))
+
+@app.get('/customers/organization/{organization_name}', response_model=List[customer_schemas.CustomerInDB])
```
[the remark this answers is not in the document]
Comment (Psami-wondah) [reply bigfastapi/customer.py:56]: Done

**Comment by Psami-wondah** [comment]
I have made some changes as requested
- added pagination for the endpoint which needed it
- named my models properly
- used org id instead of org name
- added auth for the endpoints

**Pull request merged**